### PR TITLE
[RA1 Ch06] System Hardening, password policy

### PR DESCRIPTION
**Pull request by karinesevilla**
issue #1716 (fixes partially)
To comply req sec.gen.002
Merge of sections 6.3.2.1 Function and 6.3.2.2 Software into section 6.3.2.2 Function and software
Rename section 6.3.2.1 Function into Password policy 
Content added about password policy

### CODE REVIEW

File: doc/ref_arch/openstack/chapters/chapter06.md [hunk 1]
```diff
@@ -104,10 +104,23 @@ The following rules govern create, read, update, and delete (CRUD) level access.
 ### 6.3.2 System Hardening
 All infrastructure should undergo system hardening, establishes processes to govern the hardening, and documents to cover at a minimal for the following areas:
 
-#### 6.3.2.1 Function
+#### 6.3.2.1 Password policy
+For all infrastructure components, passwords must be hardened and a strict password policy must be applied (req.sec.gen.002).
+
+Passwords must be strengthen:
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter06.md:110]:
```suggestion
Passwords must be strengthened:
```

File: doc/ref_arch/openstack/chapters/chapter06.md [hunk 2]
```diff
@@ -104,10 +104,23 @@ The following rules govern create, read, update, and delete (CRUD) level access.
...
+-	All vendors default passwords must be changed
+-	Passwords must contain at least 8 characters
+-	Passwords must contain at least one upper case letter, one lower case letter and one non-alphabetic character
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter06.md:113]: should we have: 
```suggestion
-	Passwords must contain at least one upper case letter, one lower case letter, one numeral and one special (non-alphanumeric) character.
```

File: doc/ref_arch/openstack/chapters/chapter06.md [hunk 3]
```diff
@@ -104,10 +104,23 @@ The following rules govern create, read, update, and delete (CRUD) level access.
...
+
+For passwords updates, the identity of users must be verified before providing new passwords.
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter06.md:115]:
```suggestion
For passwords updates, the identity of users must be verified before permitting a password change.
```

File: doc/ref_arch/openstack/chapters/chapter06.md [hunk 4]
```diff
@@ -104,10 +104,23 @@ The following rules govern create, read, update, and delete (CRUD) level access.
 ### 6.3.2 System Hardening
 All infrastructure should undergo system hardening, establishes processes to govern the hardening, and documents to cover at a minimal for the following areas:
 
-#### 6.3.2.1 Function
+#### 6.3.2.1 Password policy
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter06.md:107]: Should we have a policy on 
- frequency of password changes?
- restriction on the reuse of old passwords?
- restriction on use of common words, common names of places, things, name of the component, user, etc.?

File: doc/ref_arch/openstack/chapters/chapter06.md [hunk 5]
```diff
@@ -104,10 +104,23 @@ The following rules govern create, read, update, and delete (CRUD) level access.
...
+For all infrastructure components, passwords must be hardened and a strict password policy must be applied (req.sec.gen.002).
+
+Passwords must be strengthen:
+-	All vendors default passwords must be changed
+-	Passwords must contain at least 8 characters
+-	Passwords must contain at least one upper case letter, one lower case letter and one non-alphabetic character
+
+For passwords updates, the identity of users must be verified before providing new passwords.
+
+Passwords must be securely stored and transmit in an encrypted format. Password files must be stored separately from application system data.
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter06.md:117]:
```suggestion
Passwords must be encrypted at rest and in-transit. Password files must be stored separately from component and user data.
```

File: doc/ref_arch/openstack/chapters/chapter06.md (hunk 2 above)
Comment (karinesevilla) [reply doc/ref_arch/openstack/chapters/chapter06.md:113]: @pgoyal01  In my view, it was the minimal requirement for passwords for all kind of accounts. For administration, the password can be strenghtened as you proposed.
Can it be fine to add a difference between users with or without administration privileges?

File: doc/ref_arch/openstack/chapters/chapter06.md (hunk 4 above)
Comment (karinesevilla) [reply doc/ref_arch/openstack/chapters/chapter06.md:107]: @pgoyal01 As it is not testable within CNTT conformance context, we can add into best pratices recommendations. 
I am looking for a common reference document we could add here

File: doc/ref_arch/openstack/chapters/chapter06.md (hunk 5 above)
Comment (karinesevilla) [reply doc/ref_arch/openstack/chapters/chapter06.md:117]: @pgoyal01 I agree with the change proposed for the first sentence "Passwords must be encrypted at rest and in-transit.", but I prefer to keep "from application system data"

Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter06.md:117]: @karinesevilla That is fine

File: doc/ref_arch/openstack/chapters/chapter06.md (hunk 2 above)
Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter06.md:113]: @karinesevilla Yes -- that would be better

**Comment by pgoyal01** [comment]
From meeting of 2020-088-03
**Password hardening:** https://www.cisecurity.org/white-papers/ms-isac-security-primer-organizational-password-best-practices/:

1. Implement complexity rules that:
    - Allow for a minimum password length of 14 characters.
    - Force passwords to contain uppercase and lowercase letters, numbers 0 through 9, and non-alphanumeric characters.
    - Do not allow repetitive or sequential characters (e.g. ‘aaaaaa’, ‘abc123’).
    - Do not allow context-specific words, including usernames and their derivatives.
 
1. Password policies should enforce:
    - a maximum password age of between 30 and 90 days; (Comment: maybe different for users, admin, systems)
    - a minimum password age in conjunction with a password history to limit password reuse. Without a minimum password age enforcing a password history is not effective.
    - acceptance of all Unicode characters and spaces.

File: doc/ref_arch/openstack/chapters/chapter06.md (hunk 2 above)
Comment (karinesevilla) [reply doc/ref_arch/openstack/chapters/chapter06.md]: @goyal Done, 'Administration privileges accounts'  added

File: doc/ref_arch/openstack/chapters/chapter06.md (hunk 4 above)
Comment (karinesevilla) [reply doc/ref_arch/openstack/chapters/chapter06.md:107]: @goyal The CIS password policy guide is a good document for these recommendations

**Comment by karinesevilla** [comment]
@goyal
The white paper https://www.cisecurity.org/white-papers/ms-isac-security-primer-organizational-password-best-practices/ is associated with US DHS.
The CIS password policy guide https://www.cisecurity.org/white-papers/cis-password-policy-guide/ provides a set of wider recommendations

File: doc/ref_arch/openstack/chapters/chapter06.md [hunk 6]
```diff
@@ -108,14 +108,21 @@ All infrastructure should undergo system hardening, establishes processes to gov
...
+- All vendors default passwords must be changed
+- Passwords must contain at least 8 characters as a minimal value, 14 characters length passwords are recommended
+- Passwords must contain at least one upper case letter, one lower case letter and one non-alphabetic character
+- For administration privileges accounts, passwords must contain at least one upper case letter, one lower case letter, one numeral and one special (non-alphanumeric) character   
 
 For passwords updates, the identity of users must be verified before permitting a password change.
 
 Passwords must be encrypted at rest and in-transit. Password files must be stored separately from application system data.
 
+Password's composition, complexity and policy should follow the recommendations consolidated within the CIS Password Policy guide(https://www.cisecurity.org/white-papers/cis-password-policy-guide/) such as:
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter06.md:120]:
```suggestion
Password's composition, complexity and policy should follow the recommendations consolidated within the [CIS Password Policy guide](https://www.cisecurity.org/white-papers/cis-password-policy-guide/) such as:
```

File: doc/ref_arch/openstack/chapters/chapter06.md [hunk 7]
```diff
@@ -108,14 +108,21 @@ All infrastructure should undergo system hardening, establishes processes to gov
...
+- Check the password for known bad passwords(repetitive or sequential characters, dictionary words, context-specific words, previously used passwords...)
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter06.md:121]:
```suggestion
- Check the password for known bad passwords (repetitive or sequential characters, dictionary words, context-specific words, previously used passwords, etc.)
```

File: doc/ref_arch/openstack/chapters/chapter06.md [hunk 8]
```diff
@@ -108,14 +108,21 @@ All infrastructure should undergo system hardening, establishes processes to gov
...
+- Limit failed login attempts
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter06.md:122]:
```suggestion
- Limit number of failed login attempts
```

File: doc/ref_arch/openstack/chapters/chapter06.md [hunk 9]
```diff
@@ -108,14 +108,21 @@ All infrastructure should undergo system hardening, establishes processes to gov
...
+- Implement Multi-factor Authentication
+- Yearly password change or on key events e.g. indication of compromise, change of user roles, when  user leaves the organization.
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter06.md:124]:
```suggestion
- Periodic (for example, Yearly, Quarterly, etc.)  password change or on key events such as indication of compromise, change of user roles, a defined period of inactivity, when a user leaves the organization, etc..
```

**Pull request merged**